internal/pkg/statemanager/manager.go

```diff
@@ -28,6 +28,7 @@ type Manager interface {
 	AddState(key StateKey, state State) error
 	UpdateState(key StateKey, state State) error
 	DeleteState(key StateKey) error
+	GetOutputs(refParam []string) interface{}
```

Comment:
Maybe we should keep the parameter here as StateKey instead of []string, to make it more consistent?

Reply:
Maybe return two parameters, adding "error" as well. In case outputs isn't in the state.